lib/gems/pending/util/miq_object_storage.rb

```diff
@@ -0,0 +1,19 @@
+require 'util/miq_file_storage'
+
+class MiqObjectStorage < MiqFileStorage::Interface
+  attr_accessor :settings
+  attr_writer   :logger
+
+  def self.new_with_opts(opts)
+    new(opts.slice(:uri, :username, :password))
+  end
+
+  def initialize(settings)
+    raise "URI missing" unless settings.key?(:uri)
+    @settings = settings.dup
+  end
+
+  def logger
+    @logger ||= $log.nil? ? :: Logger.new(STDOUT) : $log
```

Comment:
Do we have any facility for setting the logger for this gem as a whole or is each utility kind of responsible for doing it separately? Or was everything just relying on `$log`?

Reply:
I would have to look, but I want to say it is just a cluster f...

I will report back after I do, but I want to say that this one was a copy paste from somewhere, but I don't remember off hand.

Reply:
So, we have basically "two-ish" ways we do this:

1.  We just log to `$log` directly, which is what we do in `PostgresAdmin` for example.
2.  What we are doing here.

Which is done not only in [`miq-benchmark.rb`](https://github.com/ManageIQ/manageiq-gems-pending/blob/b6ca5f0a/lib/gems/pending/util/extensions/miq-benchmark.rb#L78), but also where I guess I pulled this from (and forgot that I did) in [`MiqGenericMountSession`](https://github.com/ManageIQ/manageiq-gems-pending/blob/b6ca5f0a/lib/gems/pending/util/mount/miq_generic_mount_session.rb#L25).

So I would say we stay consistent and keep what we have here.

Reply:
> is that some syntax that I'm about to learn about?

Nope, copy pasta from `MiqGenericMountSession`...